File: src/main/java/com/gmail/borlandlp/minigamesdtools/gui/inventory/DefaultViewInventoryCreator.java
package com.gmail.borlandlp.minigamesdtools.gui.inventory;

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.Creator;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo;

import java.util.ArrayList;
import java.util.List;

@CreatorInfo(creatorId = "default_view_inventory")
public class DefaultViewInventoryCreator implements Creator {
    @Override
    public List<String> getDataProviderRequiredFields() {
        return new ArrayList<>();
    }

    @Override
    public DrawableInventory create(String ID, AbstractDataProvider dataProvider) throws Exception {
        return new DefaultViewInventory();
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gui/inventory/DefaultViewInventoryCreator.kt
package com.gmail.borlandlp.minigamesdtools.gui.inventory

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.Creator
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo
import java.util.*

@CreatorInfo(creatorId = "default_view_inventory")
class DefaultViewInventoryCreator : Creator {
    override fun getDataProviderRequiredFields(): List<String> {
        return listOf()
    }

    @Throws(Exception::class)
    override fun create(ID: String, dataProvider: AbstractDataProvider): DrawableInventory {
        return DefaultViewInventory()
    }
}